core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -578,6 +592,10 @@ private ClientFactoryOptions buildOptions() {
             return ClientFactoryOption.ADDRESS_RESOLVER_GROUP_FACTORY.newValue(addressResolverGroupFactory);
         });
 
+        final ProxyConfigSelector selector = WrappingProxyConfigSelector.of(ProxySelector.getDefault());
```

Comment:
This can be up for debate.
While users may expect DefaultProxySelector will be used out of the box, its `select` implementation isn't very efficient (even for `DIRECT` cases) affecting even normal client calls.

https://github.com/openjdk/jdk/blob/6212aea580fd018b9b7edecad9821c8208d2a12e/src/java.base/share/classes/sun/net/spi/DefaultProxySelector.java#L153

Reply:
That's a good question. It seems like JDK's default selector may not work with Armeria, either, because it doesn't understand schemes like `h2c`. Let's use `DIRECT` by default until we have a proper `ProxyConfigSelector` implementation.